### ISSUE TITLE
Transition to "Angular Strictly Typed Forms"

### ISSUE THREAD

**Issue by ianliuwk1019** [bcgov/nr-fom#346]
Starting from **Angular 14**, Angular team introduces "**Strictly Typed Forms"** to benefit it from Typescript checks and hints.
When migrating FOM from current Angular 11 to 14 and up, "ng update" (Angular CLI utility option) automatically adds this "**UntypedForm**" type to its form group, form control and any other related classes. For example:
![image.png](https://images.zenhubusercontent.com/62a9ff93d8506f14758ef03a/e682188a-2ebc-4b09-97ea-47bbb3904873)
"UntypedForm" is meant to help on transitioning to the typed world. It does not break functionality of FOM.
However, while trying initially, to transition to typed form at FOM, it does not seem easy due to the fact that FOM is using third-party **validator library** (`@rxweb/reactive-form-validators`), not the Angular built-in provided mechanism. So I feel this might take a bit more experimenting and a little more time, and since current added "UntypedForm" type does not break functionalities I create the ticket for enhancement (can be right after we migrate FOM to Angular 15).

Options can be considered:
- Somehow, with existing way of initializing a form and using reactive-form-validators at the model with annotation, code an interface of the model and use it for building the Angular **FormGroup<the-form-type>**.
Something similar to this [article](https://offering.solutions/blog/articles/2022/07/09/getting-started-with-angular-strictly-typed-reactive-forms/) (however, the article is for Angular's built-in validator).
- **RXWEB** (the validator library) does provide another package for doing [strongly typed reactive form](https://docs.rxweb.io/strongly-typed/angular-strongly-typed). I attempted to use it but did not succeed. Maybe I did not do it right or it has version combability issue. But if this can work, will be the simplest solution. We can take some time to figure it again.
- Rewrite current 'admin' reactive forms validation back to use default Angular provided form and validation setup with typed form. 

Reference:
- https://blog.angular-university.io/angular-typed-forms/
- https://offering.solutions/blog/articles/2022/07/09/getting-started-with-angular-strictly-typed-reactive-forms/
- https://docs.rxweb.io/strongly-typed/angular-strongly-typed
- https://www.npmjs.com/package/@rxweb/types
- https://stackblitz.com/edit/rxweb-types-strongly-typed-reactive-form?file=src%2Fapp%2Fapp.component.ts

**Comment by ianliuwk1019** [comment]
This is current part of the Angular upgrade for #158 as a separate subtask.
It's been fixed and currently in review for feature implementation pr#359.

**Comment by ianliuwk1019** [comment]
The third-party validator (@rxweb/reactive-form-validators) is good also for the Angular typed form.
Previously I imported the its types separately from a package dependency from https://docs.rxweb.io/strongly-typed/angular-strongly-typed and use it together with "@rxweb/reactive-form-validators" which causes error and had me confused what was going on.
However, I realized later the same package library "@rxweb/reactive-form-validators" does provide the same "type" as the typed library provides. So I use it with no problem now.
